lab_share_lib/processing/rabbit_message_processor.py

```diff
@@ -60,8 +59,15 @@ def _build_avro_encoders(self, encoder_type: str, subject: str) -> List[AvroEnco
 
     def process_message(self, headers, body):
         message = RabbitMessage(headers, body)
+        subject = message.subject
         try:
-            used_encoder = message.decode(self._build_avro_encoders(message.encoder_type, message.subject))
+            message.reader_schema_version = self._rabbit_config.message_subjects[subject].reader_schema_version
```

Comment:
So instead of having this be a property on the `RabbitMessage`, you could pass it as an argument when you call the `decode` method below.